### ISSUE TITLE
conv2d padding vs functional.pad, is there something wrong ?

### ISSUE THREAD

**Issue by vivid216** [dotnet/TorchSharp#1029]
the code:
```csharp
            var tensor = torch.zeros(1, 1, 32, 5);
            var module1 = nn.Conv2d(1, 1, (1, 3), (1, 2), padding:(0, 2));
            var module2 = nn.Conv2d(1, 1, (1, 3), (1, 2));

            var shape1 = module1.forward(tensor).shape;
            var shape2 = module2.forward(functional.pad(tensor,(0,2))).shape;
            Console.WriteLine(string.Join(',', shape1));   //output: 1,1,32,4
            Console.WriteLine(string.Join(',', shape2));   //output: 1,1,32,3
```
why Conv2d padding output shape width is 4 ?

**Comment by NiklasGustafsson** [comment]
It's how libtorch works.

In Python:

```Python
>>> tensor = torch.zeros((1,1,32,5))
>>> module1 = torch.nn.Conv2d(1, 1, (1, 3), (1, 2), padding=(0,2))
>>> module2 = torch.nn.Conv2d(1, 1, (1, 3), (1, 2))

>>> module1.forward(tensor).shape
torch.Size([1, 1, 32, 4])
>>> module2.forward(torch.nn.functional.pad(tensor, (0,2))).shape
torch.Size([1, 1, 32, 3])
```

**Comment by NiklasGustafsson** [comment]
Maybe because in the second statement, the padding is applied first, then the convolution?

Again, in Python:

```Python
>>> torch.nn.functional.pad(module2.forward(tensor), (0,2)).shape
torch.Size([1, 1, 32, 4])
```

**Comment by vivid216** [comment]
it seems Conv2d padding is padding both left and right. and functional.pad(0,2) can only padding right. 
so if i want only padding right,  i need use F.pad but not conv2d padding.

**Comment by NiklasGustafsson** [comment]
Yes, that is consistent with the documentation. The 'padding/pad' argument means different things for the two APIs:

https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html?highlight=conv2d#torch.nn.Conv2d
vs.
https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html?highlight=functional+pad#torch.nn.functional.pad

**Comment by NiklasGustafsson** [comment]
Either way, it's how libtorch works with both the Python and .NET bindings.
